cmd/sonobuoy/app/worker.go

```diff
@@ -90,7 +98,23 @@ func loadAndValidateConfig() (*plugin.WorkerConfig, error) {
 	return cfg, nil
 }
 
+// sigHandler is used to manage graceful cleanups when a TERM signal is received.
+func sigHandler() <-chan struct{} {
+	stop := make(chan struct{})
+	go func() {
+		sigc := make(chan os.Signal, 1)
+		signal.Notify(sigc)
+		sig := <-sigc
+		// The worker needs to not shutdown immediately upon receiving SIGTERM.
+		logrus.WithField("signal", sig).Info("got a signal, waiting then sending the real shutdown signal")
+		time.Sleep(pauseBeforeShutdown * time.Second)
```

Comment:
putting the pause in the sig-handler is a little weird imo.

Reply:
The alternative (and most correct) solution here is for the worker to wait until the other process is finished. I was couldn't find a nice way to do this. IPC might be a possibility, but that seemed like a lot of effort for not much gain when a sleep will do just fine.

If you have other suggestions for how to improve the robustness of this I would love to hear them since this is also not my favorite part of the code.